ports/esp32/network_lan.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 "Eric Poulsen" <eric@zyxod.com>
+ *
+ * Based on the ESP IDF example code which is Public Domain / CC0
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "network_lan.h"
+#include "py/runtime.h"
+#include "py/mphal.h"
+
+#include "eth_phy/phy.h"
+#include "eth_phy/phy_tlk110.h"
+#include "eth_phy/phy_lan8720.h"
+#include "tcpip_adapter.h"
+
+MP_DECLARE_CONST_FUN_OBJ_VAR_BETWEEN(esp_ifconfig_obj);
+
+typedef struct _lan_if_obj_t {
+    mp_obj_base_t               base;
+    int                         if_id; // MUST BE FIRST
+    bool                        initialized;
+    bool                        active;
+    uint8_t                     mdc_pin;
+    uint8_t                     mdio_pin;
+    int8_t                      phy_power_pin;
+    uint8_t                     phy_addr;
+    uint8_t                     phy_type;
+    eth_phy_check_link_func     link_func;
+    eth_phy_power_enable_func   power_func;
+} lan_if_obj_t;
+
+const mp_obj_type_t lan_if_type;
+STATIC lan_if_obj_t lan_obj = {{&lan_if_type}, ESP_IF_ETH, false, false};
+
+STATIC void phy_power_enable(bool enable) {
+    lan_if_obj_t* self = MP_OBJ_TO_PTR(&lan_obj);
+
+    if (self->phy_power_pin != -1) {
+
+        if (!enable) {
+            /* Do the PHY-specific power_enable(false) function before powering down */
+            self->power_func(false);
+        }
+
+        gpio_pad_select_gpio(self->phy_power_pin);
+        gpio_set_direction(self->phy_power_pin,GPIO_MODE_OUTPUT);
+        if(enable == true) {
```

Comment:
code style: space after if

Reply:
Also, just simply `if (enable) {`

Reply:
Same copy-paste thing.  Also, I ran `astyle` on this file to fixup that stuff.